src/lib9/open.c

```diff
@@ -318,7 +318,7 @@ dirreadmax(int fd, Dir **dp, int max)
 				return -1;
 			break;
 		}
-		if(de->d_name[de->d_namlen] != 0)
+		if(de->d_name[_D_EXACT_NAMLEN(de)] != 0)
```

Comment:
This doesn't work on AIX.
Alternatively, we could add the following line in u.h for the AIX section:

define _D_EXACT_NAMLEN(d) strlen ((d)->d_name)

Reply:
Sorry I could only test on Linux with glibc 2.3, I supposed that was a POSIX thing that could make the code work between the variants using d_namlen and d_reclen. Not sure what's the best option here. May be stick with strlen always?